pkg/providers/openstack/provider.go

```diff
@@ -141,11 +142,16 @@ func (p *Provider) serviceClientRefresh(ctx context.Context) error {
 		return fmt.Errorf("%w: password", ErrKeyUndefined)
 	}
 
+	projectID, ok := secret.Data["project-id"]
+	if !ok {
+		return fmt.Errorf("%w: project-id", ErrKeyUndefined)
+	}
+
 	// Pass in an empty string to use the default project.
-	providerClient := NewDomainScopedPasswordProvider(region.Spec.Openstack.Endpoint, string(userID), string(password), string(domainID))
+	providerClient := NewPasswordProvider(region.Spec.Openstack.Endpoint, string(userID), string(password), string(projectID))
```

Comment:
Can you add my comments in here and below RE: "the why" please, just so it's less confusing to n00bs.

Reply:
Yeah shouldn't have missed this, added comments so use is clear